optimism/src/lib.rs

```diff
@@ -50,11 +50,30 @@ pub use ramlookup::{LookupMode as RAMLookupMode, RAMLookup};
 /// we would use 3 different columns.
 pub(crate) type E<F> = Expr<ConstantExpr<F>, Column>;
 
-pub(crate) struct Circuit<const N: usize, SELECTOR, F> {
+pub struct Circuit<const N: usize, SELECTOR, F> {
+    /// The domain size of the circuit
+    pub domain_size: usize,
     /// The witness for a given selector
-    pub(crate) witness: HashMap<SELECTOR, Witness<N, Vec<F>>>,
+    pub witness: HashMap<SELECTOR, Witness<N, Vec<F>>>,
     /// The vector of constraints for a given selector
-    pub(crate) constraints: HashMap<SELECTOR, Vec<E<F>>>,
+    pub constraints: HashMap<SELECTOR, Vec<E<F>>>,
     /// The vector of lookups for a given selector
-    pub(crate) lookups: HashMap<SELECTOR, Vec<Lookup<E<F>>>>,
+    pub lookups: HashMap<SELECTOR, Vec<Lookup<E<F>>>>,
+}
+
+pub trait CircuitTrait<const N: usize, SELECTOR, F, Env> {
```

Comment:
As I mentioned in the previous PR, this to me looks much more like a generic interface for a `WitnessBuilder`. In `msm/src/ffadd/` all these functions are part of the `impl WitnessBuilderEnv`. Generally a useful abstraction, I just feel the name is confusing.

Reply:
Opened an issue to keep track of it in a follow-up https://github.com/o1-labs/proof-systems/issues/2073